### PR TITLE
Add http method to retrieve scheduler metrics

### PR DESCRIPTION
**Pull request by weiling61**
Add http method to retrieve scheduler metrics in case ssh connection does not work or is not allowed. This enables user to run tests on a node within a cluster or on master node.

### CODE REVIEW

File: clusterloader2/pkg/measurement/common/scheduler_latency.go [hunk 1]
```diff
@@ -255,10 +257,25 @@ func (s *schedulerLatencyMeasurement) sendRequestToScheduler(c clientset.Interfa
...
-		}
-		responseText = sshResult.Stdout
-	}
+                        klog.Errorf("unexpected error (code: %d) in ssh connection to master: %#v", sshResult.Code, err)
+                } else {
+                        responseText = sshResult.Stdout
+                        return responseText, nil
+                }
+                client := &http.Client{Timeout: 5 * time.Second}
+                cmd = fmt.Sprintf("http://%v:%v/metrics", masterName, ports.InsecureSchedulerPort)
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/scheduler_latency.go:266]: Sorry - I already understood the case when you're not allowed to SSH to the master - it's frequent for cloud-provider offerings.

But also - the metrics endpoint for scheduler is not exposed there too (no matter if you run inside the cluster or not).

File: clusterloader2/pkg/measurement/common/scheduler_latency.go [hunk 2]
```diff
@@ -255,10 +255,10 @@ func (s *schedulerLatencyMeasurement) sendRequestToScheduler(c clientset.Interfa
 		cmd := "curl -X " + opUpper + " http://localhost:10251/metrics"
 		sshResult, err := measurementutil.SSH(cmd, host+":22", provider)
 		if err != nil || sshResult.Code != 0 {
-			return "", fmt.Errorf("unexpected error (code: %d) in ssh connection to master: %#v", sshResult.Code, err)
-		}
-		responseText = sshResult.Stdout
-	}
+                        return "", fmt.Errorf("unexpected error (code: %d) in ssh connection to master: %#v", sshResult.Code, err)
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/scheduler_latency.go:258]: can you please fix the indentation here? apparently something got broken during your changes

File: clusterloader2/pkg/measurement/common/scheduler_latency.go [hunk 3]
```diff
@@ -96,7 +96,7 @@ func (s *schedulerLatencyMeasurement) Execute(config *measurement.Config) ([]mea
 		return nil, err
 	}
 
-	if !SSHToMasterSupported || !masterRegistered {
+	if !SSHToMasterSupported && !masterRegistered {
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/scheduler_latency.go:99]: didn't mention it before - but good catch, thanks!

File: clusterloader2/pkg/measurement/common/scheduler_latency.go (hunk 2 above)
Comment (weiling61) [reply clusterloader2/pkg/measurement/common/scheduler_latency.go]: fixed. Thanks

File: clusterloader2/pkg/measurement/common/scheduler_latency.go (hunk 3 above)
Comment (weiling61) [reply clusterloader2/pkg/measurement/common/scheduler_latency.go:99]: I signed it

**Comment by weiling61** [comment]
I signed it

**Comment by weiling61** [comment]
I signed it

________________________________

**Comment by wojtek-t** [comment]
It seem the CLA is still not signed.

Also - please squash the commits.

**Comment by weiling61** [comment]
Will close this one to fix CLA issues

**Pull request closed without merging**